korman/operators/op_mesh.py

```diff
@@ -387,6 +387,120 @@ def origin_to_bottom(obj):
+                                            default="Stand",
+                                            options=set())
+
+    def draw(self, context):
+        layout = self.layout
+        space = bpy.context.space_data
+
+        if not space.local_view:
+            box = layout.box()
+            box.label("Linking Book Name:")
+            row = box.row()
+            row.alert = not self.panel_name
+            row.prop(self, "panel_name", text="")
+            box.label("Options:")
+            row = box.row()
+            row.prop(self, "link_anim_type", text="Type")
+        else:
+            row = layout.row()
+            row.label("Warning: Operator does not work in local view mode", icon="ERROR")
+
+    def execute(self, context):
+        if context.mode == "OBJECT":
+            self.create_linkingbook_objects()
+        else:
+            self.report({"WARNING"}, "Linking Book creation only valid in Object mode")
+            return {"CANCELLED"}
+        return {"FINISHED"}
+
+
+    def create_linkingbook_objects(self):
+        bpyscene = bpy.context.scene
+        cursor_shift = mathutils.Matrix.Translation(bpy.context.scene.cursor_location)
+
+        for obj in bpy.data.objects:
+            obj.select = False
+
+        # Create Linking Panel empty
+        panel_root = bpy.data.objects.new("{}".format(self.name_stem), None)
+        panel_root.empty_draw_type = "IMAGE"
+        panel_root.empty_draw_size = 0.5
+        panel_root.empty_image_offset = (-0.5, -0.5)
+        panel_root.matrix_world = cursor_shift
+        bpy.context.scene.objects.link(panel_root)
+
+        # Create SeekPoint
+        seek_point = bpy.data.objects.new("{}".format(self.name_stem), None)
+        seek_point.empty_draw_type = "SINGLE_ARROW"
+        seek_point.rotation_euler[0] = math.pi/2
+        link_anim_offset = mathutils.Matrix.Translation(self.anim_offsets[self.link_anim_type])
+        link_anim_rot = mathutils.Matrix.Rotation(math.radians(90.0), 4, 'X')
+        seek_point.matrix_local = link_anim_offset * link_anim_rot
+        bpy.context.scene.objects.link(seek_point)
+        seek_point.plasma_object.enabled = True
+
+        # Create Region
+        clk_rgn_name = "{}_ClkRegion".format(self.name_stem)
+        clk_rgn_mesh = bpy.data.meshes.new(clk_rgn_name)
+        clk_rgn = bpy.data.objects.new(clk_rgn_name, clk_rgn_mesh)
+        clk_rgn.hide_render = True
+        clk_rgn.draw_type = "WIRE"
+
+        # Add the object into the scene.
+        bpyscene.objects.link(clk_rgn)
+        clk_rgn.plasma_object.enabled = True
+
+        # Construct the clickable region and assign it to the blender mesh.
+        bm = bmesh.new()
+        clk_rgn_size = 6.0
+        bmesh.ops.create_cube(bm, size=(1.0), matrix=(mathutils.Matrix.Scale(clk_rgn_size, 4)))
+        bm.to_mesh(clk_rgn_mesh)
+        bm.free()
```

Comment:
I've seen this same anti-pattern in two PRs now. Maybe it would be best to write a contextlib helper so we can do something like:
```python
with utils.bmesh_object("My Fancy Object") as clk_rgn, bm:
        clk_rgn_size = 6.0
        bmesh.ops.create_cube(bm, size=(1.0), matrix=(mathutils.Matrix.Scale(clk_rgn_size, 4)))
```

Implementation would look something like pseudo-python:
```python
@contextmanager
def bmesh_object(name):
    mesh = bpy.data.meshes.new(name)
    obj = bpy.data.objects.new(name, mesh)
    obj.draw_type = "WIRE"
    bpy.context.scene.objects.link(obj)
    bm = bmesh.new()
    try:
        yield obj, bm
    else:
        bm.to_mesh(mesh)
    finally:
        bm.free()
```

Reply:
It's used in more places than that!
I welcome a better solution, so this looks good to me.